### PR TITLE
feat: Add support for batching configuration

### PR DESCRIPTION
**Pull request by losalex**
Fixes #[953](https://github.com/googleapis/java-logging/issues/953), #[730](https://github.com/googleapis/java-logging/issues/730), #[705](https://github.com/googleapis/java-logging/issues/705) ☕️

### CODE REVIEW

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingOptions.java [hunk 1]
```diff
@@ -108,6 +118,8 @@ public LoggingOptions build() {
   protected LoggingOptions(Builder builder) {
     super(LoggingFactory.class, LoggingRpcFactory.class, builder, new LoggingDefaults());
     this.autoPopulateMetadataOnWrite = builder.autoPopulateMetadataOnWrite;
+    this.batchingSettings =
+        builder.batchingSettings == null ? null : builder.batchingSettings.toBuilder().build();
```
Comment (minherz) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingOptions.java:122]: Is the call `batchingSettings.toBuilder().build()` necessary here? In the builder code (line 107) this transformation is already executed.

File: google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/GrpcLoggingRpc.java [hunk 2]
```diff
@@ -169,18 +169,25 @@ public Void apply(UnaryCallSettings.Builder<?, ?> builder) {
...
-                  .setFlowControlSettings(
-                      oldBatchSettings
-                          .getFlowControlSettings()
-                          .toBuilder()
-                          .setLimitExceededBehavior(LimitExceededBehavior.Block)
-                          .build())
-                  .build());
+              batchingSettings != null
+                  ? batchingSettings
+                  : oldBatchSettings
```
Comment (minherz) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/GrpcLoggingRpc.java:182]: nit: I think this is an opportunity to refactor the code. Consider renaming `oldBatchSettings` to `defaultBatchSettings`.

**Review by minherz — COMMENTED** [review]
Left a couple of remarks.
I did not review README and testing though :slightly_smiling_face:

File: google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/GrpcLoggingRpc.java (hunk 2 above)
Comment (losalex) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/spi/v2/GrpcLoggingRpc.java:182]: Sure, makes sense

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingOptions.java (hunk 1 above)
Comment (losalex) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingOptions.java:122]: Oh, good catch. I believe I will leave constructor as is and remove it from `setBatchingSettings()` - this way it is more clear

**Review by minherz — APPROVED** [review]
lgtm

**Review by daniel-sanche — APPROVED** [review]

**Pull request merged**